src/arith/iter_affine_map.cc

```diff
@@ -351,31 +367,45 @@ class IterMapRewriter : public ExprMutator {
     std::vector<IterSplitExpr> iters;
     iters.reserve(expr->args.size());
     // canonicalize the expression
+    // find the base scale first
+    IntImm base_scale(nullptr);
```

Comment:
Use Optional<IntImm> to better express our intent that base_scale is not defined in the begining.  This can make things better for future cases when we make IntImm not nullable

Alternatively, we can make base_index int64, and use base_index == -1 to determine its value